core/lib/dal/src/events_web3_dal.rs

```diff
@@ -111,7 +128,11 @@ impl EventsWeb3Dal<'_, '_> {
         }
     }
 
-    fn build_get_logs_where_clause(&self, filter: &GetLogsFilter) -> (String, u8) {
+    fn build_get_logs_where_clause(
+        &self,
+        filter: &GetLogsFilter,
+        skip_transfer_event: bool,
```

Comment:
I would suggest turning this boolean value into an `enum` something like

```rust
enum ApiMode {
  Legacy,
  Modern,
 }
 ```
 
For improved readability (which is which?) and future-proofing (agianst situations where we will be running more than two apis (we will).